v2/spanner-change-streams-to-sharded-file-sink/src/test/java/com/google/cloud/teleport/v2/templates/SpannerChangeStreamToGcsSimpleIT.java

```diff
@@ -0,0 +1,453 @@
+      artifactsShardB =
+          gcsResourceManager.listArtifacts("output/testShardB/", Pattern.compile(".*\\.txt$"));
+      artifactsShardC =
+          gcsResourceManager.listArtifacts("output/testShardC/", Pattern.compile(".*\\.txt$"));
+
+      // Ideally both the mutations written to spanner per shard will commit within 10 seconds.
+      // But that does not guarantee that they will be in the same file, since they can commit
+      // within 1 second interval boundary
+      if (artifactsShardB.size() >= 1 && artifactsShardC.size() >= 1) {
+        break;
+      }
+    }
+    assertThatArtifacts(artifactsShardB).hasFiles();
+    assertThatArtifacts(artifactsShardC).hasFiles();
+    // checks that any of the artifact has the given content
+    assertThatArtifacts(artifactsShardB).hasContent("SingerId\\\":\\\"2");
+    assertThatArtifacts(artifactsShardB).hasContent("SingerId\\\":\\\"3");
+    assertThatArtifacts(artifactsShardC).hasContent("SingerId\\\":\\\"4");
+    assertThatArtifacts(artifactsShardC).hasContent("SingerId\\\":\\\"5");
+  }
+
+  @Test
+  public void testAllDatatypes() throws IOException, java.lang.InterruptedException {
+    // Construct a ChainedConditionCheck with below stages.
+    // 1. Wait for the metadata table to have the start time of reader job
+    // 2. Write a record with
+    // 3. Wait on GCS to have the files
+    // 4. Match the PK in GCS with the PK written to Spanner
+    ChainedConditionCheck conditionCheck =
+        ChainedConditionCheck.builder(
+                List.of(
+                    SpannerRowsCheck.builder(
+                            spannerMetadataResourceManager, "spanner_to_gcs_metadata")
+                        .setMinRows(1)
+                        .setMaxRows(1)
+                        .build()))
+            .build();
+    // Wait for conditions
+    PipelineOperator.Result result =
+        pipelineOperator()
+            .waitForCondition(createConfig(jobInfo, Duration.ofMinutes(10)), conditionCheck);
+    // Assert Conditions
+    assertThatResult(result).meetsConditions();
+    // Perform writes to Spanner
+    writeSpannerDataForAllDatatypes();
+    // Assert file present in GCS with the needed data
+    assertFileContentsInGCSForAllDatatypes();
+  }
+
+  private void writeSpannerDataForAllDatatypes() {
+    // Write a single record to Spanner for logical shard : testD
+    Mutation m =
+        Mutation.newInsertOrUpdateBuilder("sample_table")
+            .set("id")
+            .to(1)
+            .set("varchar_column")
+            .to("abc")
+            .set("migration_shard_id")
+            .to("testShardD")
+            .set("tinyint_column")
+            .to(1)
+            .set("text_column")
+            .to("aaaaaddd")
+            .set("year_column")
+            .to("2023")
+            .set("smallint_column")
+            .to(22)
+            .set("bigint_column")
+            .to(12345678910L)
```

Comment:
We have also added tests with lowest possible values and highest possible values in Forward migration ITs. It can be added in subsequent PRs.